File: Data Structures & Algorithms/count-number-of-islands/submission-1.py
class Solution:
    def numIslands(self, grid: List[List[str]]) -> int:
        islands = 0
        for r in range(len(grid)):
            for c in range(len(grid[0])):
                if grid[r][c] == '1':
                    nbrs = deque([(r,c)])
                    islands += 1
                    while nbrs:
                        cr, cc = nbrs.popleft()
                        for dr, dc in [(0,1),(0,-1),(1,0),(-1,0)]:
                            nr, nc = cr + dr, cc + dc
                            if 0 <= nr < len(grid) and 0 <= nc < len(grid[0]) and grid[nr][nc] == '1':
                                grid[nr][nc] = '0'
                                nbrs.append((nr, nc))
        return islands

File: Data Structures & Algorithms/is-subsequence/submission-1.cs
public class Solution {
    public bool IsSubsequence(string s, string t) {
        int i = 0;
        int j = 0;

        while (i < t.Length && j < s.Length) {
            if (t[i] == s[j]) {
                j++;
            }

            i++;
        }

        if (s.Length == j) {
            return true;
        }
    
    return false;
    }
}